src/main/java/com/example/webapp/Service/DepartmentService.java

```diff
@@ -2,58 +2,65 @@
 
 import com.example.webapp.Model.Department;
 import com.example.webapp.Repository.DepartmentRepository;
+import com.example.webapp.dto.DepartmentDTO;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 
 @Service
 public class DepartmentService {
-@Autowired
-DepartmentRepository departmentRepository;
-    public ResponseEntity<List<Department>> getAll(){
-        try{
-            List<Department> departmentList = new ArrayList<>();
-            departmentRepository.findAll().forEach(departmentList::add);
-            if(departmentList.isEmpty()){
-                return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-            }
-            return new ResponseEntity<>(departmentRepository.findAll(), HttpStatus.OK);
-        } catch (Exception e){
-            return new ResponseEntity<>(null, HttpStatus.INTERNAL_SERVER_ERROR);
+    @Autowired
+    DepartmentRepository departmentRepository;
+
+    public List<Department> getAll() {
+        List<Department> departmentList = new ArrayList<>();
+        departmentRepository.findAll().forEach(department -> departmentList.add(department));
+        if (departmentList.isEmpty()) {
+            return Collections.emptyList();
         }
+        return departmentList;
     }
 
-    public ResponseEntity<Department> getById(long id){
+    public Department getById(long id) {
         Optional<Department> dep = departmentRepository.findById(id);
-        if(dep.isPresent()){
-            return new ResponseEntity<>(dep.get(), HttpStatus.OK);
-        }
-        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        return dep.orElse(null);
     }
 
-    public ResponseEntity<Department> add(Department department){
-        try{
-            Department department1 = departmentRepository.save(department);
-            return new ResponseEntity<>(department1, HttpStatus.CREATED);
-        } catch (Exception e) {
-            return new ResponseEntity<>(null, HttpStatus.INTERNAL_SERVER_ERROR);
-        }
+    public DepartmentDTO add(DepartmentDTO department) {
```

Comment:
finalize all parameter arguments

Reply:
change to ```public DepartmentDTO add(DepartmentDTO departmentDTO)``` ?

Reply:
public DepartmentDTO add(final DepartmentDTO department) if department remains unmodifiable